services/blockchain-connector/tests/functional/download.test.js

```diff
@@ -0,0 +1,99 @@
+
+	beforeAll(async () => {
+		// Create test directory
+		await mkdir(directoryPath, { recursive: true });
+	});
+
+	afterAll(async () => {
+		await rm(directoryPath, { recursive: true, force: true });
+	});
+
+	afterEach(async () => {
+		await rm(`${directoryPath}/genesis_block.json`);
+		await rm(`${directoryPath}/apiJson.json`);
+	});
+
+	it('should download and extract tar file -> valid url', async () => {
+		const [{ genesisBlockUrl }] = config.networks.LISK.filter(network => network.name === 'testnet');
+		const filePath = `${directoryPath}/genesis_block.json`;
+		expect(exists(filePath)).resolves.toBe(false);
+		await downloadAndExtractTarball(genesisBlockUrl, directoryPath);
+		expect(exists(filePath)).resolves.toBe(true);
+	});
+
+	it('should download and extract tar file -> invalid url', async () => {
+		const url = 'https://downloads.lisk.com/lisk/testnet/genesis_block_invalid.json';
+		expect(downloadAndExtractTarball(url, directoryPath)).rejects.toThrow();
+	});
+
+	it('should download JSON file -> valid url', async () => {
+		const url = 'https://raw.githubusercontent.com/LiskHQ/lisk-service/development/services/gateway/apis/http-version3/swagger/apiJson.json';
+		const filePath = `${directoryPath}/apiJson.json`;
+		expect(exists(filePath)).resolves.toBe(false);
+		await downloadJSONFile(url, filePath);
+		expect(exists(filePath)).resolves.toBe(true);
+	});
+
+	it('should download JSON file -> invalid url', async () => {
+		const url = 'https://downloads.lisk.com/lisk/testnet/genesis_block.json';
+		const filePath = `${directoryPath}/genesis_block.json`;
+		expect(downloadJSONFile(url, filePath)).rejects.toThrow();
+	});
+
+	it('should download and unzip file -> valid url', async () => {
+		const [{ genesisBlockUrl }] = config.networks.LISK.filter(network => network.name === 'testnet');
+		const filePath = `${directoryPath}/genesis_block.json`;
+		expect(exists(filePath)).resolves.toBe(false);
+		await downloadAndUnzipFile(genesisBlockUrl, filePath);
+		expect(exists(filePath)).resolves.toBe(true);
+	});
+
+	it('should download and unzip file -> invalid url', async () => {
+		const url = 'https://downloads.lisk.com/lisk/testnet/service-core-snapshot-invalid.sql.gz';
+		const filePath = `${directoryPath}/service-core-snapshot-invalid.sql`;
+		expect(downloadAndUnzipFile(url, filePath)).rejects.toThrow();
+	});
+
+	it('should download file -> valid url', async () => {
+		const url = 'https://raw.githubusercontent.com/LiskHQ/lisk-service/development/services/gateway/apis/http-version3/swagger/apiJson.json';
+		const filePath = `${directoryPath}/apiJson.json`;
+		expect(exists(filePath)).resolves.toBe(false);
+		await downloadFile(url, directoryPath);
+		expect(exists(filePath)).resolves.toBe(true);
+	});
+
+	it('should download file -> invalid url', async () => {
+		const url = 'https://downloads.lisk.com/lisk/testnet/genesis_block.json';
+		expect(downloadFile(url, directoryPath)).rejects.toThrow();
+	});
+});
```

Comment:
1. can we group the functions by describe? 
2. can we add input cases of empty string, null, undefined, not empty invalid string where applicable?

Please consider the above points of other test files as well.